tests/src/pancake_test.cpp

```diff
@@ -35,19 +35,23 @@ int main(){
         keys.push_back(std::to_string(i));
     }
     std::vector<std::string> values = std::vector<std::string>(keys.size(), dummy);
-    void *arguments[3];
+    auto id_to_client = std::make_shared<thrift_response_client_map>();
+    void *arguments[4];
     arguments[0] = malloc(sizeof(distribution *));
```

Comment:
Again, malloc's are unused. I'd avoid mallocs whenever possible. The cardinal rule of C/C++ is that you must not have malloc's without corresponding frees, lest you want memory leaks.